flink-core/src/main/java/org/apache/flink/util/InstantiationUtil.java

```diff
@@ -169,6 +169,7 @@ public ClassLoaderObjectInputStream(InputStream in, ClassLoader classLoader) thr
 		scalaTypes.add("scala.Tuple2$mcDJ$sp");
 		scalaTypes.add("scala.Tuple2$mcDI$sp");
 		scalaTypes.add("scala.Tuple2$mcDD$sp");
+		scalaTypes.add("scala.Enumeration$ValueSet");
```

Comment:
This seems like fixing one specific case that was discovered. There is probably a ton of unsupported cases. I would suggest that we either:
  - See why this is a problem in the first place (classes should not be serialized into savepoints or should not be in the serializers any more).
  - Do not support Scala version switches with savepoint compatibility (due to scala)